### ISSUE TITLE
Add `clusterctl move` to CI

### ISSUE THREAD

**Issue by jichenjc** [kubernetes-sigs/cluster-api-provider-openstack#777]
/kind feature

**Describe the solution you'd like**
[A clear and concise description of what you want to happen.]

we create cluster and run conformance test , but the cluster is not moved
better to add this `move` action before we run conformance test

**Comment by jichenjc** [comment]
Take 
https://logs.openlabtesting.org/logs/76/776/95893c32f7d950cb0cd39284eb3fe90e7e900ba0/check/cluster-api-provider-openstack-current-acceptance-test-v1.18/19d03f7/job-output.txt.gz

as example,
```
2021-03-09 02:55:39.953620 | ubuntu-bionic-large | capi-quickstart-control-plane-69gh4     openstack:///05d9415d-d4a5-47c4-b928-b94d91907337   Running   v1.18.6
2021-03-09 02:55:39.953649 | ubuntu-bionic-large | capi-quickstart-md-0-64587867bb-krrwp   openstack:///acedce55-dec2-468e-95e5-64f31e6fdd4c   Running   v1.18.6
2021-03-09 02:55:40.061809 | ubuntu-bionic-large | Running tests...
2021-03-09 02:55:40.061878 | ubuntu-bionic-large | Running tests in parallel
2021-03-09 02:55:40.415473 | ubuntu-bionic-large | Conformance test: not doing test setup.
2021-03-09 02:55:42.493159 | ubuntu-bionic-large | Running Suite: Kubernetes e2e suite
```

we created a cluster and we start the test, but we should consider move objects from 
kind cluster to workload cluster (so that we can simulate more real world for something like machine api handling etc)

**Comment by sbueringer** [comment]
I plan to implement more advanced e2e tests soon (after the Prow PR is merged). Could it be that this is one of the scenarios tested by other providers (I saw something like node scale up/down, but I'm not sure if they are also testing clusterctl move)

**Comment by sbueringer** [comment]
I think after the move to the new e2e tests we could take a look if other providers are doing it and how. 
Although I think this goes against the pattern of 99% of the current e2e tests in CAPI right now.

The usually work like this:
* start *one* kind cluster and deploy *one* CAPI/CAPO control plane
* then you run several tests concurrently which test e.g. different cluster topologies

If we would now move the *one* control plane in one of those tests the others won't work anymore. I wouldn't say 
there is no way to do this, but it's not trivial. Also could be that one of the other providers already found a good way to do this.

**Comment by sbueringer** [comment]
e2e tests are now merged in CAPO. I also found a e2e test in CAPI which is similar to our e2e test. So if we want to test this we should do it very similar to how it's implemented here:
https://github.com/kubernetes-sigs/cluster-api/blob/master/test/e2e/self_hosted.go#L77-L168

**Comment by jichenjc** [comment]
let me give a try later on, thanks for the reminder

**Comment by sbueringer** [comment]
Of course. Just wanted to give an update on the e2e tests overall and the test I found in CAPI. No stress :)

**Comment by tobiasgiese** [comment]
@jichenjc is this still an open issue? Are you on it?
Otherwise I can take a look, if it's okay for you.

**Comment by jichenjc** [comment]
I am not working on it now, feel free to work on it 
@sbueringer and I had discussion on this before and seems this is something should be covered by capi itself and
nice to have at capo .. so it's optional but worthy to add if we have time/resource to work on, thanks for take a look at this~

**Comment by tobiasgiese** [comment]
Sure, I'll take a look at it :)

/assign
